sklearn/utils/estimator_checks.py

```diff
@@ -1457,6 +1457,7 @@ def param_filter(p):
             # true for mixins
             return
         params = estimator.get_params()
+        estimator.set_params(**params)
```

Comment:
Please put a comment in here to describe what this is testing.

Also: Should we do further checking that `get_params()` returns the same thing before and after this statement.